core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchReaderIterator.java

```diff
@@ -329,6 +331,8 @@ private void processFailures(Map<KeyExtent,List<Range>> failures, ResultReceiver
 
     // since the first call to binRanges clipped the ranges to within a tablet, we should not get only
     // bin to the set of failed tablets
+    if (!locator.isValid())
+      locator = new TimeoutTabletLocator(TabletLocator.getLocator(instance, new Text(table)), timeout);
```

Comment:
Using an AtomicReference might be more straightforward and more localized of a change (avoids any potential lock contention on TSBRI.class)